### PR TITLE
Prepare data object context and tabs

### PR DESCRIPTION
**Pull request by markus-moser**
Implement general data object context and element context handling + the following data object tabs:
- Properties
- Schedule
- Dependencies
- Notes & Events
- Tags
- Workflow
- Versions Tab preparation - dummy components for single and comparison view are used for now

### CODE REVIEW

File: assets/js/src/core/app/config/services/index.ts [hunk 1]
```diff
@@ -73,6 +79,12 @@ container.bind(serviceIds['Asset/Editor/ArchiveTabManager']).to(ArchiveTabManage
 container.bind(serviceIds['Asset/Editor/UnknownTabManager']).to(UnknownTabManager).inSingletonScope()
 container.bind(serviceIds['Asset/MetadataTypeProvider/MetadataTypeRegistry']).to(MetadataTypeRegistry).inSingletonScope()
 
+// Data Objects
+container.bind(serviceIds['DataObject/Editor/ComponentRegistry']).to(ComponentRegistryService).inSingletonScope()
```
Comment (vin0401) [inline assets/js/src/core/app/config/services/index.ts:83]: To be honest I'm not sure why this one was bound initially to the assets. 
It should be generic for all case.

**Review by vin0401 — CHANGES_REQUESTED** [review]
Just one small thing with the component registry. 
Apart from that it's super awesome to see the first steps for the objects! :)

File: assets/js/src/core/app/config/services/index.ts (hunk 1 above)
Comment (markus-moser) [reply assets/js/src/core/app/config/services/index.ts:83]: Should it be in the element context or totally global for all kind of components?

Comment (vin0401) [reply assets/js/src/core/app/config/services/index.ts:83]: I would make it totally global. So far I would not see the benefit to limit it to a specific area. 
The components are registered via a descriptive identifier, anyway.

Comment (markus-moser) [reply assets/js/src/core/app/config/services/index.ts:83]: Fixed in #563

**Pull request merged**